### PR TITLE
Adding AWS server setup scripts

### PR DESCRIPTION
**Pull request by aniket03**
Adding file to setup Server on EC2
Adding file to cluster nodes and set HA policy

### CODE REVIEW

File: set_RMQ_EC2.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+##Should run on local machine
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: Refer to https://www.python.org/dev/peps/pep-0008
Comments should start with one `#`

File: set_RMQ_EC2.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+## And ssh into your AWS accounts then start rabbitmq-server
+
+import paramiko
+
+hostnames = [
+	'ec2-54-173-2-90.compute-1.amazonaws.com'		##Your instance's public name
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: Inline comments should be separated by two spaces. 
```py
a = b  # dummy comment
```

File: set_RMQ_EC2.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	
+]
+## since gonna be same for each host
+k = paramiko.RSAKey.from_private_key_file("/home/aniket/Downloads/rabbitmqkey.pem")
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: `k` doesnt tell me what it is. So variable names should be readable.

Comment (ketanbhatt) [reply set_RMQ_EC2.py]: `"/home/aniket/Downloads/rabbitmqkey.pem"`
things like these can be set as variables and then variables can be used in the code later. 

So later if I want to change the values of these params, I dont have to read the full code, just change the value of the params

File: set_RMQ_EC2.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+##To set up RMQ-server up and running on each node
+for host in hostnames:
+	print "Setting RabbitMQ on host", host
+	## To establish ssh session with the reqd server
+	ssh = paramiko.SSHClient()
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: `ssh_client`

File: set_RMQ_EC2.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+	ssh.connect(hostname=host, username='ubuntu', pkey=k)
+	transport = ssh.get_transport()
+	session = transport.open_session()
+	session.set_combine_stderr(True)
+	session.get_pty()
+
+	## To execute the commands
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: This comment is not needed

File: set_RMQ_EC2.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	commands =[
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: `commands_to_be_executed`

File: set_RMQ_EC2.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+		'sudo -i apt-get install rabbitmq-server -y',
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: Maybe add inline comments telling what each command is doing.

File: set_RMQ_EC2.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+		'sudo rabbitmq-plugins enable rabbitmq_management',
+		'sudo -i rabbitmq-server -detached',
+		'sudo -i rabbitmqctl stop_app',
+		'sudo -i rabbitmqctl reset',
+		'sudo -i rabbitmqctl start_app',
+		'sudo -i rabbitmqctl cluster_status',
+	]
+
+	## To make a single command
+	single_command = ""
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: Making a list and changing that to string is faster than appending to strings.

Comment (ketanbhatt) [reply set_RMQ_EC2.py]: Also, based on the usage, you could just do `single_command = '\n'.join(commands)`

Comment (ketanbhatt) [reply set_RMQ_EC2.py]: `single_command` --> `command_to_execute`

File: set_RMQ_EC2.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+	session.exec_command(single_command)
+	stdin = session.makefile('wb', -1)
+	stdout = session.makefile('rb', -1)
+	shell_lines = stdout.readlines()
+	for s in shell_lines:
+		print s
+	ssh.close()
+
+'''
+Chain of Events
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: Move this to some other file if you want.

Comment (ketanbhatt) [reply set_RMQ_EC2.py]: Also: There are no multiline comments in python yet.
https://www.python.org/dev/peps/pep-0008/#block-comments

Also, fun: https://twitter.com/vinayakcoder/status/670178989500071937

File: set_RMQ_EC2.py [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+	## To execute the commands
+	commands =[
+		'sudo -i apt-get install rabbitmq-server -y',
+		'sudo rabbitmq-plugins enable rabbitmq_management',
+		'sudo -i rabbitmq-server -detached',
+		'sudo -i rabbitmqctl stop_app',
+		'sudo -i rabbitmqctl reset',
+		'sudo -i rabbitmqctl start_app',
+		'sudo -i rabbitmqctl cluster_status',
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py:33]: This is not required.

File: set_RMQ_cluster_AMI.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
+import paramiko
+
+hostnames = [
```
Comment (ketanbhatt) [inline set_RMQ_cluster_AMI.py]: This can be made using master and slaves.

File: set_RMQ_cluster_AMI.py [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+	'ec2-54-173-2-90.compute-1.amazonaws.com',		##Master(original)
+	'ec2-54-234-142-144.compute-1.amazonaws.com'	##slave(image)
+
+]
+
+master = 'ec2-54-173-2-90.compute-1.amazonaws.com'
+slaves = [
+	'ec2-54-234-142-144.compute-1.amazonaws.com'
+]	##Since there can be multiple slaves
```
Comment (ketanbhatt) [inline set_RMQ_cluster_AMI.py]: Comment not required

File: set_RMQ_cluster_AMI.py [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+pr_ip = {}	## Dict to store private ip of each host
```
Comment (ketanbhatt) [inline set_RMQ_cluster_AMI.py]: Better naming required

File: set_RMQ_cluster_AMI.py [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+	ssh = paramiko.SSHClient()
+	ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+	ssh.connect(hostname=host, username='ubuntu', pkey=k)
+	transport = ssh.get_transport()
+	session = transport.open_session()
+	session.set_combine_stderr(True)
+	session.get_pty()
+
+	## To execute the command to get host name
+	session.exec_command("hostname")
```
Comment (ketanbhatt) [inline set_RMQ_cluster_AMI.py]: Can I directly read the output of this command? There has to be a way

File: set_RMQ_cluster_AMI.py [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+	stdin = session.makefile('wb', -1)
```
Comment (ketanbhatt) [inline set_RMQ_cluster_AMI.py]: Files not required too.

File: set_RMQ_EC2.py [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+	single_command = ""
+	for i in range(len(commands)):
+		if i < len(commands)-1:
+			single_command=single_command + (commands[i]+"\n")
+		else:
+			single_command=single_command + (commands[i])
+
+	print single_command
+	session.exec_command(single_command)
+	stdin = session.makefile('wb', -1)
```
Comment (ketanbhatt) [inline set_RMQ_EC2.py]: Not required

File: set_RMQ_cluster_AMI.py [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+	stdout = session.makefile('rb', -1)
+	shell_lines = stdout.readlines()
+	pr_ip[host] = shell_lines[0][:len(shell_lines[0])-2]
+	##so as to remove last 2 chars
+	ssh.close()
+
+## To cluster the nodes using join_cluster 
+## will be launched by slave
+for slave in slaves:
+	ssh = paramiko.SSHClient()
```
Comment (ketanbhatt) [inline set_RMQ_cluster_AMI.py]: Since setting up is same, lets abstract this to a method

File: set_RMQ_cluster_AMI.py [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+ssh = paramiko.SSHClient()
+ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+ssh.connect(hostname=master, username='ubuntu', pkey=k)
+transport = ssh.get_transport()
+session = transport.open_session()
+session.set_combine_stderr(True)
+session.get_pty()
+
+## To execute the command to get host name
+session.exec_command("sudo rabbitmqctl set_policy ha-all \"\" \'{\"ha-mode\":\"all\",\"ha-sync-mode\":\"automatic\"}\'")
```
Comment (ketanbhatt) [inline set_RMQ_cluster_AMI.py]: This command is not readable. 
Lets define the policy in a dict and then do a json.dumps() to get a string for it.

**Review by ketanbhatt — CHANGES_REQUESTED** [review]
Good work

**Review by aniket03 — COMMENTED** [review]
Fine

**Comment by ketanbhatt** [comment]
@aniket03 so we generally do the review until the reviewer says that further review is not required. 

Please take care next time 😄

**Pull request merged**